### PR TITLE
docs: Use --recursive alias to --recurse-submodules

### PR DESCRIPTION
**Pull request by matthewfeickert**
* Amends PR #55

### CODE REVIEW

**Comment by matthewfeickert** [comment]
Sorry, should have just copy and pasted instead of thinking I can type correctly after midnight.

File: README.md
Comment (jpivarski) [inline README.md:102]:
```suggestion
git clone --recursive https://github.com/scikit-hep/fastjet.git
```

I don't know if `--recurse-submodules` is equivalent to `--recursive`, but I've only ever used the latter.

Also, isn't it more common to quote `https://` URLs than `git@`?

**Review by jpivarski — COMMENTED** [review]
This is a comment because I'm suggesting a change that might not be necessary.

File: README.md
Comment (matthewfeickert) [reply README.md:102]: > I don't know if `--recurse-submodules` is equivalent to `--recursive`, but I've only ever used the latter.

Yeah. I had recently seeing more docs using `--recurse-submodules` and it is the official CLI argument

```console
$ git clone --help
```
```
SYNOPSIS
       git clone [--template=<template_directory>]
                 [-l] [-s] [--no-hardlinks] [-q] [-n] [--bare] [--mirror]
                 [-o <name>] [-b <name>] [-u <upload-pack>] [--reference <repository>]
                 [--dissociate] [--separate-git-dir <git dir>]
                 [--depth <depth>] [--[no-]single-branch] [--no-tags]
                 [--recurse-submodules[=<pathspec>]] [--[no-]shallow-submodules]
                 [--[no-]remote-submodules] [--jobs <n>] [--sparse] [--] <repository>
                 [<directory>]
... 
       --recurse-submodules[=<pathspec]
           After the clone is created, initialize and clone submodules within based on the provided pathspec. If no pathspec is provided, all submodules are initialized and cloned. This option can
           be given multiple times for pathspecs consisting of multiple entries. The resulting clone has submodule.active set to the provided pathspec, or "." (meaning all submodules) if no
           pathspec is provided.

           Submodules are initialized and cloned using their default settings. This is equivalent to running git submodule update --init --recursive <pathspec> immediately after the clone is
           finished. This option is ignored if the cloned repository does not have a worktree/checkout (i.e. if any of --no-checkout/-n, --bare, or --mirror is given)
```

BUT it also seems [that `--recursive` is an alias to `--recurse-submodules`](https://explainshell.com/explain?cmd=git+clone+--recursive) so let's go with the one that uses less typing!

> Also, isn't it more common to quote `https://` URLs than `git@`?

Probably. As the section was labeled "Installation For Developers" I assumed that you're going to have a bad time if you're not already using SSH keys trying to develop, so thought I might as well put in something that would be more focused there. But there's certainly not any harm in using `https`.

Comment (aryan26roy) [reply README.md]: So are we going with https? If so should I just make the change? @matthewfeickert @jpivarski

Comment (jpivarski) [reply README.md]: I think what Aryan meant by "for developers" is that ordinary users would pip install it, once the bugs are fixed in the binary wheel building process. There ought to be an "installation" section just above the "installation for developers," though.

Comment (aryan26roy) [reply README.md]: @jpivarski Isn't the "installation" section at the top enough?

Comment (jpivarski) [reply README.md]: Let's go with `--recursive` and HTTPS just because it's what we've been using on other Scikit-HEP projects that I know of (in particular, Awkward Array). Expressing it different ways makes it seem like there are important differences, but there's nothing in particular that makes fastjet need SSH keys, so that would be a source of confusion.

Comment (matthewfeickert) [reply README.md]: > So are we going with https? If so should I just make the change?

Oh that's totally your call. Whatever you and @jpivarski suggest go for it. :+1:

Comment (jpivarski) [reply README.md]: > @jpivarski Isn't the "installation" section at the top enough?

It is! I missed it because they're far from each other. Don't worry about that—it's fine as is.

Comment (aryan26roy) [reply README.md]: I have made the change and after it has passed the tests I'll merge it and close this issue/PR.

**Review by aryan26roy — APPROVED** [review]

**Pull request merged**
